libkrun/Makefile

```diff
@@ -1,8 +1,28 @@
 LIBKRUNTIME_CFLAGS =	-Wall -shared -fPIC
 COMMON_CFLAGS = -Wall -pedantic -std=gnu99
+SUDO = sudo
 
-ifeq ("${TRAVIS}", "true")
-	COMMON_CFLAGS += -DTRAVIS
+# Sadly this tool requires root
+VIRT_WHAT = ${SUDO} /usr/sbin/virt-what
+
+# Detect if we are virtualised (Linux only for now XXX)
+ifeq ($(shell uname -s),Linux)
+	VIRTUALISED = 0  # default off
+
+	VIRT_FACTS = $(shell ${VIRT_WHAT})
+	ifneq ("${VIRT_FACTS}","")
+		VIRTUALISED = 1
+	endif
+
+	ifeq ("${TRAVIS}","true")
```

Comment:
So that we know whether or not to build libkrun with or without performance counter support. I suppose that could have been runtime checked, but it could have gotten in the way of taking readings.